src/test/java/com/wikia/webdriver/testcases/discussions/CreatingReplyTests.java

```diff
@@ -8,73 +8,185 @@
+    int expandedEditorHeight = replyCreator.getEditorHeight();
+    Assertion.assertTrue(expandedEditorHeight > originalEditorHeight, "Reply editor should expand.");
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
   }
 
-  private void userOnMobileMustBeLoggedInToUseReplyCreator() {
-    ReplyCreatorMobile replyCreator = new PostDetailsPage().openDefaultPost().getReplyCreatorMobile();
+  @Test(groups = USER_ON_DESKTOP_GROUP)
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DiscussionsConstants.DESKTOP_RESOLUTION)
+  public void userOnDesktopCanCreateReplyOnPostDetailsPage() {
+    final PostDetailsPage page = new PostDetailsPage().open(createPostAsUserRemotely().getId());
+    final ReplyCreator replyCreator = page.getReplyCreatorDesktop();
 
+    assertThatUserCanCreateReply(page, replyCreator);
+  }
+
+  @Test(groups = USER_ON_DESKTOP_GROUP)
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DiscussionsConstants.DESKTOP_RESOLUTION)
+  public void userOnDesktopShouldFollowPostWhenAddedReplyOnPostDetailsPage() {
+    final PostDetailsPage page = new PostDetailsPage().open(createPostAsUserRemotely().getId());
+    final ReplyCreator replyCreator = page.getReplyCreatorDesktop();
+
+    assertThatPostIsFollowedWhenReplyAdded(page, replyCreator);
+    assertThatPostCanBeUnfollowed(page);
+  }
+
+  // Testing methods
+
+  private void userOnDesktopMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.REGISTER_PAGE);
+  }
+
+  private void anonymousUserOnReplyEditorClickIsRedirectedTo(final ReplyCreator replyCreator, final String urlFragment) {
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickOkButtonInSignInDialog();
-
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickSignInButtonInSignInDialog();
+    Assertion.assertTrue(driver.getCurrentUrl().contains(urlFragment));
+  }
+
+  private void userOnMobileMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.JOIN_PAGE);
+  }
+
+  private PostEntity.Data createPostAsUserRemotely() {
+    return DiscussionsOperations.using(User.USER, driver).cratePostWithUniqueData();
+  }
+
+  private void assertThatUserCanCreateReply(PostDetailsPage page, ReplyCreator replyCreator) {
+    replyCreator.click().clickGuidelinesReadButton();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
+    replyCreator.add(TextGenerator.createUniqueText());
+    Assertion.assertTrue(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_ACTIVE_MESSAGE);
+
+    replyCreator.clearText();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
```

Comment:
`assertThatUserCanCreateReply` should check just that - I think that lines 147 thorugh 155 are reduntant. They could be a part of another tests - that would check whether:

- clicking on guidelines button would make reply button disabled
- adding a text makes reply button enabled
- clearing a text would make reply button disabled

The main difference is that I don't think we should run these 3 additional scenarios for all user types, because they check general behavior of adding a reply. So this would make the main tests that use `assertThatUserCanCreateReply` faster.